### PR TITLE
Feature/29738 user supplied bvn for orgunit

### PR DESCRIPTION
**Pull request by jgj-magenta-aps**
https://redmine.magenta-aps.dk/issues/29738

- [ ] This is a trivial change (and the rest of the list can be ignored)

**Please ensure the following is true:**

- [X] The application can be built and installed without errors
- [X] The feature/bugfix has been tested manually (if manually testable)
- [ ] Tests have been written/updated for my change
- [x] The documentation has been updated
    - [x] The documentation builds without warnings or errors
- [x] Release notes have been updated
- [x] The corresponding Redmine ticket has been set to `Needs review`
    - [x] The issue has a link to this PR

**If applicable:**

- [X] Changes have been made to the UI
    - [X] Screenshots of relevant changes to UI added to PR
- [x] The API/data model has been changed/expanded
    - [x] Redmine tickets have been created to update dependent systems
- [ ] The feature introduces changes relevant to the deployment process
    - [ ] Redmine tickets have been created to update deployment process
- [ ] This change affects the development workflow.
    - [ ] I have notified other developers by email or chat.

Sådan ser ændringen ud

nyt-felt-user_key

Jeg har smoketestet lidt - henholdsvis med autowash (5th element):

auto-wash-user_key

og med udfyldt bvn

supplied_user_key

### CODE REVIEW

File: backend/mora/service/orgunit.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def prepare_create(self, req):
         )
 
         unitid = util.get_uuid(req, required=False) or str(uuid.uuid4())
-        bvn = util.checked_get(req, mapping.USER_KEY, unitid)
+        bvn = util.checked_get(req, mapping.USER_KEY, unitid) or unitid
```
Comment (danchr) [inline backend/mora/service/orgunit.py]: Passing `null` or `None` here should trigger the fallback if needed; disregarding `""` — i.e. the empty string — is unusual and inconsistent with our other APIs. (See also https://github.com/os2mo/os2mo/commit/bc54c1bc6e28.)

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue [hunk 2]
```diff
@@ -13,6 +13,12 @@
         required
       />
 
+      <mo-input-text
+        :label="$t('input_fields.user_key')"
+        :placeholder="$t('input_fields.user_key_placeholder')"
+        v-model="entry.user_key"
```
Comment (danchr) [inline frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: Should we make this a required field?
```suggestion
        v-model="entry.user_key"
        required
```

File: backend/mora/service/orgunit.py (hunk 1 above)
Comment (jgj-magenta-aps) [reply backend/mora/service/orgunit.py]: Fedt - den er jo helt for sig selv. Kan man fjerne et enkelt commit fra pullrequestet?

File: frontend/src/locales/da/input_fields.json [hunk 3]
```diff
@@ -17,6 +17,8 @@
   "name": "Navn",
   "new_name": "Nyt navn",
   "org_unit_type": "@:shared.org_unit_type",
+  "user_key": "Brugervendt nøgle",
```
Comment (danchr) [inline frontend/src/locales/da/input_fields.json]: I think we should check with Alex or perhaps a customer how they refer to this; “brugervendt nøgle” seems overly technical to me. Perhaps “Kort navn” or (ew) “Enhedsnummer”?

File: backend/mora/service/orgunit.py (hunk 1 above)
Comment (danchr) [reply backend/mora/service/orgunit.py]: Prøv `git rebase -i`?

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue (hunk 2 above)
Comment (jgj-magenta-aps) [reply frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: Det ved jeg ikke. jeg tror vi bliver nødt til at spørge Alex - og du er jo tættest på ham for tiden? Med mindre du da har sendt ham på vej imod KBH.

File: frontend/src/locales/da/input_fields.json (hunk 3 above)
Comment (jgj-magenta-aps) [reply frontend/src/locales/da/input_fields.json]: Agree! - han har fået skærmbilledet med placeholder og hele molevitten lige op i raketten

File: backend/mora/service/orgunit.py (hunk 1 above)
Comment (jgj-magenta-aps) [reply backend/mora/service/orgunit.py]: Nu vil det bryde med løsningen at acceptere den tomme streng, meningen er jo netop at man lader brugeren vælge om han vil angive den eller ikke. Ellers skal man jo lave noget frontend, der gør at man ikke overfører den, hvis den ikke er udfyldt - er det ikke lidt bagvendt med model etc?

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue (hunk 2 above)
Comment (jgj-magenta-aps) [reply frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: Jævnfør anden kommentar: Det er kun nogle gange, de vil angive den, så required tror jeg ikke er godt.

File: frontend/src/locales/da/input_fields.json (hunk 3 above)
Comment (jgj-magenta-aps) [reply frontend/src/locales/da/input_fields.json]: Han var glad for skærmbilledet - og Brugervendt nøgle er jo det vi kalder den? Eller er det Enhedsnummer nu? Hvad så med medarbejdere - det er jo samme koncept?

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue [hunk 4]
```diff
@@ -85,6 +91,10 @@ export default {
      */
     entry: {
       handler (newVal) {
+        if (newVal.user_key === undefined || newVal.user_key === ""){
+          newVal.user_key = null; 
+        }
+        console.log(newVal.user_key);
```
Comment (cmoesgaard) [inline frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: Remove this log statement

File: frontend/src/locales/da/input_fields.json [hunk 5]
```diff
@@ -17,6 +17,8 @@
   "name": "Navn",
   "new_name": "Nyt navn",
   "org_unit_type": "@:shared.org_unit_type",
+  "user_key": "Enhedsnummer",
+  "user_key_placeholder": "Udfyld eller auto",
```
Comment (cmoesgaard) [inline frontend/src/locales/da/input_fields.json]: "Udfyld eller auto" is not an especially helpful message

File: frontend/src/locales/da/input_fields.json [hunk 6]
```diff
@@ -17,6 +17,8 @@
   "name": "Navn",
   "new_name": "Nyt navn",
   "org_unit_type": "@:shared.org_unit_type",
+  "user_key": "Enhedsnummer",
```
Comment (cmoesgaard) [inline frontend/src/locales/da/input_fields.json]: Rename this to something more specific to org unit user keys - we wouldn't want _every_ potential user_key field to be called "Enhedsnummer"

File: NEWS.rst [hunk 7]
```diff
@@ -4,6 +4,10 @@ In development
 API changes
 -----------
 
+``/service/ou/create``:
+
+If an empty user_key is supplied, the uuid of the orgunit is used
```
Comment (danchr) [inline NEWS.rst]: This is no longer true, is it?

File: NEWS.rst [hunk 8]
```diff
@@ -12,6 +16,9 @@ To bypass this check, specify ``force=1``.
 Bug fixes
 ---------
 
+* #29738: user_key can be entered in UI for organisational units.
```
Comment (danchr) [inline NEWS.rst]: I think this is a new feature :)

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue (hunk 4 above)
Comment (danchr) [inline frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: Efterladt aflusningskode?
```suggestion
```

File: NEWS.rst (hunk 7 above)
Comment (jgj-magenta-aps) [reply NEWS.rst]: No - it will be if NO userkey - will fix it

Comment (jgj-magenta-aps) [reply NEWS.rst]: I removed - it is now standard, isn't it

File: NEWS.rst (hunk 8 above)
Comment (jgj-magenta-aps) [reply NEWS.rst]: moved

File: frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue (hunk 4 above)
Comment (jgj-magenta-aps) [reply frontend/src/components/MoEntry/MoOrganisationUnitEntry.vue]: done

File: frontend/src/locales/da/input_fields.json (hunk 5 above)
Comment (jgj-magenta-aps) [reply frontend/src/locales/da/input_fields.json]: Better than nothing - Alex loved it :-)

File: backend/mora/service/orgunit.py (hunk 1 above)
Comment (jgj-magenta-aps) [reply backend/mora/service/orgunit.py]: Done - null is now enforced when not filled

File: frontend/src/locales/da/input_fields.json (hunk 5 above)
Comment (cmoesgaard) [reply frontend/src/locales/da/input_fields.json]: Alright then. If it works for El Jefe, it works for me.

**Pull request merged**